src/ossia-max/src/ZeroconfMinuitListener.cpp

```diff
@@ -95,68 +106,60 @@ std::vector<std::string> ZeroconfMinuitListener::m_embryo_devices;
     }
   }
 
-  void ZeroconfMinuitListener::instanceRemoved(const std::string& instance)
-  {
-    std::lock_guard<std::mutex> lock(ZeroconfMinuitListener::m_mutex);
-    auto it = ossia::find_if(m_devices, [&](const auto& d) {
-      return d.name == instance;
-    });
-
-    if (it != m_devices.end())
-    {
-      m_zombie_devices.push_back(it);
-    }
-  }
-
-  ossia::net::minuit_connection_data* ZeroconfMinuitListener::find_device(
+  std::shared_ptr<ossia::net::generic_device> ZeroconfMinuitListener::find_device(
       const std::string& instance)
   {
-    std::lock_guard<std::mutex> lock(ZeroconfMinuitListener::m_mutex);
+    std::lock_guard<std::mutex> lock(ZeroconfMinuitListener::s_mutex);
     {
-      auto it = ossia::find_if(m_devices, [&](const auto& d) {
-        return d.name == instance + " Minuit server";
+      auto it = ossia::find_if(s_devices, [&](const auto& d) {
+        return d->get_name() == instance + " Minuit server";
       });
 
-      if (it != m_devices.end())
+      if (it != s_devices.end())
       {
-        return &(*it);
+        return *it;
       }
-      m_mutex.unlock();
+      s_mutex.unlock();
     }
-    return nullptr;
+    return {};
 
   }
 
   void ZeroconfMinuitListener::browse()
   {
-    ZeroconfMinuitListener::m_mutex.lock();
-
-    // FIXME we should take receiving order into account
-    // a device that is first removed, then added
-    // will currently be removed
-    for(const auto& s : m_embryo_devices)
-    {
-      process_new_devices(s);
-    }
-    m_embryo_devices.clear();
+    ZeroconfMinuitListener::s_mutex.lock();
```

Comment:
I'd go for std::lock_guard here as this could cause the mutex to always be locked if something underneath throws